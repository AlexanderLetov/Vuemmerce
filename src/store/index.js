import Vue from 'vue';
import Vuex from 'vuex';
import initalState from '../initial-state.json';
import search from './modules/search';

Vue.use(Vuex);

export default new Vuex.Store({
  strict: true,

  state: initalState,

  getters: {
    getCategoryById: state => id => {
      return state.categories.find(category => category.id == id);
    },
    getProductsList: state => {
      return state.products;
    },
    productsAdded: state => {
      return state.products.filter(el => {
        return el.isAddedToCart;
      });
    },
    productsAddedToFavourite: state => {
      return state.products.filter(el => {
        return el.isFavourite;
      });
    },
    getProductById: state => id => {
      return state.products.find(product => product.id == id);
    },
    getProductImages: state => id => {
      return state.products.find(product => product.id == id).images;
    },
    isUserLoggedIn: state => {
      return state.userInfo.isLoggedIn;
    },
    isUserSignedUp: state => {
      return state.userInfo.isSignedUp;
    },
    getUserName: state => {
      return state.userInfo.name;
    },
    getUserEmail: state => {
      return state.userInfo.email;
    },
    getUserAddress: state => {
      return state.userInfo.address;
    },
    getUserPassword: state => {
      return state.userInfo.password;
    },
    getAvatar: state => {
      return state.userInfo.avatarBinary;
    },
    isLoginModalOpen: state => {
      return state.systemInfo.openLoginModal;
    },
    isSignupModalOpen: state => {
      return state.systemInfo.openSignupModal;
    },
    isCheckoutModalOpen: state => {
      return state.systemInfo.openCheckoutModal;
    },
    quantity: state => {
      return state.products.quantity;
    },
    getNewsById: state => id => {
      return state.news.find(newItem => newItem.id == id);
    }
  },

  actions: {
    addToCartAction ({ commit }, id) {
      return new Promise((resolve, reject) => {
          commit('addToCart', id);
          commit('setAddedBtn', id);
          resolve();
      })
    }
  },

  mutations: {
    addToCart: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isAddedToCart = true;
        }
      });
    },
    setAddedBtn: (state, data) => {
      state.products.forEach(el => {
        if (data.id === el.id) {
          el.isAddedBtn = data.status;
        }
      });
    },
    removeFromCart: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isAddedToCart = false;
        }
      });
    },
    removeProductsFromFavourite: state => {
      state.products.filter(el => {
        el.isFavourite = false;
      });
    },
    isUserLoggedIn: (state, isUserLoggedIn) => {
      state.userInfo.isLoggedIn = isUserLoggedIn;
    },
    isUserSignedUp: (state, isSignedUp) => {
      state.userInfo.isSignedUp = isSignedUp;
    },
    setUserName: (state, name) => {
      state.userInfo.name = name;
    },
    saveAvatar: (state, avatarBinary) => {
      state.userInfo.avatarBinary = avatarBinary;
    },
    setHasUserSearched: (state, hasSearched) => {
      state.userInfo.hasSearched = hasSearched;
    },
    setUserName: (state, name) => {
      state.userInfo.name = name;
    },
    setUserEmail: (state, email) => {
      state.userInfo.email = email;
    },
    setUserAddress: (state, address) => {
      state.userInfo.address = address;
    },
    setUserPassword: (state, password) => {
      state.userInfo.password = password;
    },
    setProductTitleSearched: (state, titleSearched) => {
      state.userInfo.productTitleSearched = titleSearched;
    },
    showLoginModal: (state, show) => {
      state.systemInfo.openLoginModal = show;
    },
    showSignupModal: (state, show) => {
      state.systemInfo.openSignupModal = show;
    },
    showCheckoutModal: (state, show) => {
      state.systemInfo.openCheckoutModal = show;
    },
    addToFavourite: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isFavourite = true;
        }
      });
    },
    removeFromFavourite: (state, id) => {
      state.products.forEach(el => {
        if (id === el.id) {
          el.isFavourite = false;
        }
      });
    },
    quantity: (state, data) => {
      state.products.forEach(el => {
        if (data.id === el.id) {
          el.quantity = data.quantity;
        }
      });
    }
  },

  modules: {
    search
  }
})
