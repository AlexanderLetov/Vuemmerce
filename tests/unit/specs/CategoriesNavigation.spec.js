import { mount } from '@vue/test-utils';
import { createRenderer } from 'vue-server-renderer';
import component from '../../../src/components/categories_nav/CategoriesNavigation.vue';
import store from '../../../src/store';
import router from '../../../src/router';

const localVue = global.newVueInstance();

describe('CategoriesNavigation.vue', async () => {
	it('mounts successfully', () => {
		const wrapper = mount(component, {
			localVue,
			router,
			store
		});
		expect(wrapper.exists()).toBe(true);
	});

	it('snapshot renderer test', () => {
		const renderer = createRenderer();
		const wrapper = mount(component, {
			localVue,
			router,
			store,
			propsData: { id: 1 }
		});
		renderer.renderToString(wrapper.vm, (err, str) => {
			if (err) throw new Error(err);
			expect(str).toMatchSnapshot();
		});
	});
});
