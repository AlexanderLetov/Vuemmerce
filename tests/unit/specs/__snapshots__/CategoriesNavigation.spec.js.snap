// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesNavigation.vue snapshot renderer test 1`] = `
<div class="menu">
  <p class="menu-label">Categories</p>
  <ul class="menu-list">
    <div><a href="/category-products/1" class="details">
        <li><a class="is-active"><span class="icon is-small"><i class="mdi mdi-information-outline"></i></span> <span>Category 1</span></a>
          <!---->
        </li>
      </a></div>
    <div><a href="/category-products/2" class="details">
        <li><a><span class="icon is-small"><i class="mdi mdi-information-outline"></i></span> <span>Category 2</span></a>
          <!---->
        </li>
      </a></div>
    <div><a href="/category-products/3" class="details">
        <li><a><span class="icon is-small"><i class="mdi mdi-information-outline"></i></span> <span>Category 3</span></a>
          <!---->
        </li>
      </a></div>
  </ul>
</div>
`;
